src/control/server/mgmt_client.go

```diff
@@ -41,37 +41,75 @@ const (
 )
 
 type (
-	mgmtSvcClientCfg struct {
+	// GrpcClientCfgHelper provides config access.
+	GrpcClientCfgHelper interface {
+		GetConfig() *grpcClientCfg
+		SetConfig(grpcClientCfg)
+		LeaderAddress() (string, error)
+	}
+	// grpcClientCfgHelper implements GrpcClientCfgHelper.
+	grpcClientCfgHelper struct {
+		cfg grpcClientCfg
+	}
+	// GrpcClient provides a subset of MgmtSvcClient for operations between gRPC servers.
+	GrpcClient interface {
+		GrpcClientCfgHelper
+		Join(context.Context, *mgmtpb.JoinReq) (*mgmtpb.JoinResp, error)
+		PrepShutdown(context.Context, string, mgmtpb.RanksReq) (*mgmtpb.RanksResp, error)
+		Stop(context.Context, string, mgmtpb.RanksReq) (*mgmtpb.RanksResp, error)
+		Start(context.Context, string, mgmtpb.RanksReq) (*mgmtpb.RanksResp, error)
+		Query(context.Context, string, mgmtpb.RanksReq) (*mgmtpb.RanksResp, error)
+	}
```

Comment:
From a naming standpoint, "GrpcClient" is a bit vague. How about "SystemClient", or maybe "HarnessClient"? Just something to clarify what kind of client this is.